### PR TITLE
nagstamon: 3.14.0 -> 3.16.2

### PR DESCRIPTION
**Pull request by liberodark**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by jwillikers — APPROVED** [review]

File: pkgs/tools/misc/nagstamon/default.nix [hunk 1]
```diff
@@ -1,21 +1,26 @@
...
+
   build-system = with pythonPackages; [ setuptools ];
-  dependencies = with pythonPackages; [ configparser pyqt6 psutil requests
+  dependencies = with pythonPackages; [ arrow configparser pyqt6 psutil requests
      beautifulsoup4 keyring requests-kerberos lxml dbus-python python-dateutil pysocks ];
 
+  # Use wrapQtAppsHook to wrap the app properly
+  nativeBuildInputs = [ qt6.wrapQtAppsHook ];
+
   meta = with lib; {
```
Comment (FliegendeWurst) [inline pkgs/tools/misc/nagstamon/default.nix:24]: Set platforms = linux. Doesn't build on Mac.

Comment (khaneliman) [reply pkgs/tools/misc/nagstamon/default.nix:24]: Platforms should represent what upstream supports. If it doesn't build we should mark it as a badPlatforms. The error looks like it just might need tweaking to fix. 

'''bash

    >               ^^^^^^^^^^^^^^^^^
       > NameError: name 'bdist_rpm_options' is not defined. Did you mean: 'bdist_mac_options'?

Comment (liberodark) [reply pkgs/tools/misc/nagstamon/default.nix:24]: Sorry but I don't quite understand what you expect from me.
Sorry the goal of this PR is to fix this application which has been broken for several versions of nixos.

Comment (liberodark) [reply pkgs/tools/misc/nagstamon/default.nix:24]: Need to add just : platforms = platforms.linux; ?

File: pkgs/tools/misc/nagstamon/default.nix [hunk 2]
```diff
@@ -1,25 +1,58 @@
...
+  buildInputs = [
+    qt6.qtmultimedia
+    qt6.qtbase
+    qt6.qtsvg
+  ];
+
+  # Correction : buildSystem -> nativeBuildInputs
+  nativeBuildInputs = with pythonPackages; [
+    setuptools
+    qt6.wrapQtAppsHook # Déplacé ici car c'est un outil de build
```
Comment (Artturin) [inline pkgs/tools/misc/nagstamon/default.nix:29]: English please, the comment is unnecessary.

File: pkgs/tools/misc/nagstamon/default.nix [hunk 3]
```diff
@@ -1,25 +1,58 @@
...
-  build-system = with pythonPackages; [ setuptools ];
-  dependencies = with pythonPackages; [ configparser pyqt6 psutil requests
-     beautifulsoup4 keyring requests-kerberos lxml dbus-python python-dateutil pysocks ];
+  buildInputs = [
+    qt6.qtmultimedia
+    qt6.qtbase
+    qt6.qtsvg
+  ];
+
+  # Correction : buildSystem -> nativeBuildInputs
```
Comment (Artturin) [inline pkgs/tools/misc/nagstamon/default.nix:26]: Don't rename, those attr names are what the docs use. They probably want to do something under the hood with them in the future.

Keep the wrapQtAppsHook in nativeBuildInputs

**Review by Artturin — CHANGES_REQUESTED** [review]

File: pkgs/top-level/all-packages.nix [hunk 4]
```diff
@@ -9417,6 +9417,7 @@ with pkgs;
 
   nagstamon = callPackage ../tools/misc/nagstamon {
     pythonPackages = python3Packages;
+    qt6 = kdePackages;
```
Comment (Artturin) [inline pkgs/top-level/all-packages.nix:9420]: Why? You're not using anything specific to that set, use qt6Packages directly

File: pkgs/tools/misc/nagstamon/default.nix (hunk 3 above)
Comment (SuperSandro2000) [reply pkgs/tools/misc/nagstamon/default.nix:26]: We are only using build-system when pyproject is set to true

File: pkgs/tools/misc/nagstamon/default.nix [hunk 5]
```diff
@@ -1,25 +1,55 @@
...
 pythonPackages.buildPythonApplication rec {
   pname = "nagstamon";
-  version = "3.14.0";
+  version = "3.16.2";
 
   src = fetchurl {
     url = "https://github.com/HenriWahl/Nagstamon/archive/refs/tags/v${version}.tar.gz";
-    sha256 = "sha256-9RxQ/rfvoyjSUsY4tmAkBdVQqZYi3X6PBzQYFIeenzA=";
+    sha256 = "sha256-QuXs0muY0CvEJFB7ehMhvcQG4MByp3ySEmr7qB1enEk=";
   };
```
Comment (GaetanLepage) [inline pkgs/tools/misc/nagstamon/default.nix:15]: Why not using `fetchFromGitHub` ?

**Comment by GaetanLepage** [comment]
Ideally, the formatting changes should be isolated in a separate commit.

**Comment by liberodark** [comment]
> Ideally, the formatting changes should be isolated in a separate commit.

Hi,
Can you make exception about this ?
Is easy to read all changes and have fix this broken app.
So I made all the requested changes.

Best Regards

File: pkgs/tools/misc/nagstamon/default.nix [hunk 6]
```diff
@@ -1,25 +1,57 @@
...
-  version = "3.14.0";
+  version = "3.16.2";
 
-  src = fetchurl {
-    url = "https://github.com/HenriWahl/Nagstamon/archive/refs/tags/v${version}.tar.gz";
-    sha256 = "sha256-9RxQ/rfvoyjSUsY4tmAkBdVQqZYi3X6PBzQYFIeenzA=";
+  src = fetchFromGitHub {
+    owner = "HenriWahl";
+    repo = "Nagstamon";
+    rev = "v${version}";
```
Comment (GaetanLepage) [inline pkgs/tools/misc/nagstamon/default.nix:15]:
```suggestion
    rev = "refs/tags/v${version}";
```
This prevents branch/tag mismatch.

File: pkgs/tools/misc/nagstamon/default.nix [hunk 7]
```diff
@@ -1,25 +1,57 @@
-{ lib, fetchurl, pythonPackages }:
+{
+  lib,
+  fetchFromGitHub,
+  pythonPackages,
```
Comment (GaetanLepage) [inline pkgs/tools/misc/nagstamon/default.nix:4]:
```suggestion
  python3Packages,
```

File: pkgs/tools/misc/nagstamon/default.nix [hunk 8]
```diff
@@ -1,25 +1,57 @@
...
+    qt6.qtsvg
+  ];
+
+  nativeBuildInputs = [ qt6.wrapQtAppsHook ];
+
   build-system = with pythonPackages; [ setuptools ];
-  dependencies = with pythonPackages; [ configparser pyqt6 psutil requests
-     beautifulsoup4 keyring requests-kerberos lxml dbus-python python-dateutil pysocks ];
+
+  dependencies = with pythonPackages; [
```
Comment (GaetanLepage) [inline pkgs/tools/misc/nagstamon/default.nix:32]:
```suggestion
  dependencies = with python3Packages; [
```

File: pkgs/tools/misc/nagstamon/default.nix [hunk 9]
```diff
@@ -1,25 +1,57 @@
...
 
+  buildInputs = [
+    qt6.qtmultimedia
+    qt6.qtbase
+    qt6.qtsvg
+  ];
+
+  nativeBuildInputs = [ qt6.wrapQtAppsHook ];
+
   build-system = with pythonPackages; [ setuptools ];
```
Comment (GaetanLepage) [inline pkgs/tools/misc/nagstamon/default.nix:30]:
```suggestion
  build-system = with python3Packages; [ setuptools ];
```

**Review by GaetanLepage — CHANGES_REQUESTED** [review]

File: pkgs/top-level/all-packages.nix [hunk 10]
```diff
@@ -9416,7 +9416,7 @@ with pkgs;
   naabu = callPackage ../tools/security/naabu { };
 
   nagstamon = callPackage ../tools/misc/nagstamon {
-    pythonPackages = python3Packages;
+    qt6 = qt6Packages;
```
Comment (GaetanLepage) [inline pkgs/top-level/all-packages.nix:9419]: Is there a reason for not directly using `qt6Packages` in the derivation inputs ?

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 349714`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>nagstamon</li>
    <li>nagstamon.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>nagstamon</li>
    <li>nagstamon.dist</li>
  </ul>
</details>

---
### `x86_64-darwin`

---
### `aarch64-darwin`

**Review by GaetanLepage — APPROVED** [review]
LGTM !

**Comment by liberodark** [comment]
Hi, 

Also if is possible to backport to 24.11 ?

Best Regards

File: pkgs/by-name/na/nagstamon/package.nix [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+  # Test assumes darwin
+  doCheck = false;
+
+  buildInputs = [
+    qt6Packages.qtmultimedia
+    qt6Packages.qtbase
+    qt6Packages.qtsvg
+  ];
+
+  nativeBuildInputs = [ qt6Packages.wrapQtAppsHook ];
```
[the remark this answers is not in the document]
Comment (liberodark) [reply pkgs/by-name/na/nagstamon/package.nix:28]: Nop because is Qt app

[the remark this answers is not in the document]
Comment (FliegendeWurst) [reply pkgs/by-name/na/nagstamon/package.nix:28]: I think the issue is that both the Python wrapper and the Qt wrapper will apply.

Therefore you should add
```
  dontWrapQtApps = true;

  preFixup = ''
    makeWrapperArgs+=("''${qtWrapperArgs[@]}")
  '';
```
to make sure it is only wrapped once.

**Comment by GaetanLepage** [comment]
I was not able to launch the application successfully on wayland:
```
❯ /nix/store/vd1d202vlfxhagxsx4hh6pfwlmafzd5i-nagstamon-3.16.2/bin/nagstamon.py 
[Errno 2] No such file or directory: '/home/gaetan/.nagstamon/nagstamon.pid'
qt.qpa.plugin: Could not find the Qt platform plugin "wayland" in ""
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: linuxfb, xcb, minimal, offscreen, minimalegl, vkkhrdisplay, vnc, eglfs.

fish: Job 1, '/nix/store/vd1d202vlfxhagxsx4hh…' terminated by signal SIGABRT (Abort)
```

**Comment by liberodark** [comment]
Hi,

I dont have wayland sorry but i think this can be fixed.

Best Regards

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 349714`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>nagstamon</li>
    <li>nagstamon.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>nagstamon</li>
    <li>nagstamon.dist</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>nagstamon</li>
    <li>nagstamon.dist</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>nagstamon</li>
    <li>nagstamon.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Fails on darwin with
```
Executing setuptoolsBuildPhase
Traceback (most recent call last):
  File "/private/tmp/nix-build-nagstamon-3.16.2.drv-0/source/nix_run_setup", line 8, in <module>
    exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\\r\\n', '\\n'), __file__, 'exec'))
  File "setup.py", line 162, in <module>
    bdist_rpm=bdist_rpm_options)
              ^^^^^^^^^^^^^^^^^
NameError: name 'bdist_rpm_options' is not defined. Did you mean: 'bdist_mac_options'?
```

**Comment by liberodark** [comment]
Yes but have disable darwin plateform.

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 349714`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>nagstamon</li>
    <li>nagstamon.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>nagstamon</li>
    <li>nagstamon.dist</li>
  </ul>
</details>

---
### `x86_64-darwin`

---
### `aarch64-darwin`

**Comment by GaetanLepage** [comment]
@Artturin could you please have another look at this PR ? It is good to go as far as I am concerned.

File: pkgs/by-name/na/nagstamon/package.nix [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+  };
+
+  build-system = with python3Packages; [ setuptools ];
+
+  buildInputs = [
+    qt6Packages.qtmultimedia
+    qt6Packages.qtbase
+    qt6Packages.qtsvg
+    qt6Packages.qtwayland
+  ];
```
Comment (Artturin) [inline pkgs/by-name/na/nagstamon/package.nix:26]:
```suggestion
  nativeBuildInputs = [ qt6Packages.wrapQtAppsHook ];

  buildInputs = [
    qt6Packages.qtmultimedia
    qt6Packages.qtsvg
  ];
```

qtbase and qtwayland are propagated by the wrapper

Comment (GaetanLepage) [reply pkgs/by-name/na/nagstamon/package.nix]: Thanks for the tip. Fixed.

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 349714`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>nagstamon</li>
    <li>nagstamon.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>nagstamon</li>
    <li>nagstamon.dist</li>
  </ul>
</details>

---
### `x86_64-darwin`

---
### `aarch64-darwin`

**Comment by GaetanLepage** [comment]
cc @Artturin for final review

**Review by Artturin — APPROVED** [review]

**Pull request merged**